lib/services/services_linux.c

```diff
@@ -866,20 +867,30 @@ services__configuration_error(svc_action_t *op, bool is_fatal)
 void
 services__handle_exec_error(svc_action_t * op, int error)
 {
+    const char *name = op->opaque->exec;
+
+    if (name == NULL) {
+        name = op->agent;
+        if (name == NULL) {
+            name = op->id;
+        }
+    }
+
     switch (error) {   /* see execve(2), stat(2) and fork(2) */
         case ENOENT:   /* No such file or directory */
         case EISDIR:   /* Is a directory */
         case ENOTDIR:  /* Path component is not a directory */
         case EINVAL:   /* Invalid executable format */
         case ENOEXEC:  /* Invalid executable format */
             services__set_result(op, services__not_installed_error(op),
-                                 PCMK_EXEC_NOT_INSTALLED,
-                                 "%s", pcmk_rc_str(error));
+                                 PCMK_EXEC_NOT_INSTALLED, "%s: %s",
```

Comment:
Sometimes the result strings are like a sentence that is self-explanatory when printed standalone.
Sometimes it would need a bit of context to know what it is about - like in this case.
Is there a strict strategy?

Reply:
The execution status provides context, e.g. here the end result will be something like "Not installed: /path/to/blah: No such file or directory". The more detailed messages are useful mainly when the execution status is just "Error".